aws-android-sdk-s3/src/main/java/com/amazonaws/mobileconnectors/s3/transferutility/TransferUtility.java

```diff
@@ -597,6 +603,48 @@ public TransferObserver upload(String key, File file, ObjectMetadata metadata, C
         return upload(getDefaultBucketOrThrow(), key, file, metadata, cannedAcl, listener);
     }
 
+    /**
+     * Starts uploading the inputStream to the <b>default</b> bucket, using the given key.
+     *
+     * @param key           The key in the specified bucket by which to store the new object.
+     * @param inputStream   The input stream to upload.
+     * @return A TransferObserver used to track upload progress and state
+     */
+    public TransferObserver upload(String key, InputStream inputStream) throws IOException {
+        return upload(key, inputStream, UploadOptions.builder().build());
+    }
+
+    /**
+     * Starts uploading the inputStream to the given bucket, using the given key.
+     *
+     * @param key           The key in the specified bucket by which to store the new object.
+     * @param inputStream   The input stream to upload.
+     * @param options       An UploadOptions which hold all of the optional parameters
+     *                      i.e. bucket, metadata, cannedAcl and transferListener.
+     * @return A TransferObserver used to track upload progress and state
+     */
+    public TransferObserver upload(String key, InputStream inputStream, UploadOptions options) throws IOException {
+        File file = writeInputStreamToFile(inputStream);
+        TransferObserver observer = upload(
+                options.getBucket() != null ? options.getBucket() : getDefaultBucketOrThrow(),
+                key,
+                file,
+                options.getMetadata() != null ? options.getMetadata() : new ObjectMetadata(), options.getCannedAcl(),
+                options.getTransferListener()
+        );
+        deleteTempFile(observer);
+        return observer;
+    }
+
+    synchronized public void deleteTempFile(TransferObserver observer) {
```

Comment:
Doesn't need to be `public`, right?

Reply:
In fact, we want to hide this detail if possible, so i think it's best left `private`. If a customer happens to have created a file that also begins with "aws-s3" and calls this method not knowing what it does, they will end up losing data.

nit: i think accessor keyword should come first so it should probably be `private synchronized void`